GH-36621: [C++] Add documentation for ACERO_ALIGNMENT_HANDLING (#36622)
apache/arrow: docs/source/cpp/env_vars.rst

process (for example, when the Arrow C++ DLL is loaded), so you cannot assume
that changing their value later will have an effect.

.. envvar:: ACERO_ALIGNMENT_HANDLING

   Arrow C++'s Acero module performs computation on streams of data.  This
   computation may involve a form of "type punning" that is technically
   undefined behavior if the underlying array is not properly aligned.  On
   most modern CPUs this is not an issue, but some older CPUs may crash or

>    suffer poor performance.  For this reason it is recommended that all
>    incoming array buffers are properly aligned.

Reviewer: ```suggestion
   suffer poor performance.  For this reason it is recommended that all
   incoming array buffers are properly aligned, but some data sources
   such as :ref:`Flight <flight-rpc>` may produce unaligned buffers.
```

   The value of this environment variable controls what will happen when

>    Acero detects an unaligned buffer.

Reviewer: ```suggestion
   Acero detects an unaligned buffer:
```

>    - ``warn`` a warning is emitted
>    - ``ignore`` nothing, alignment checking is disabled
>    - ``reallocate`` the buffer is reallocated to a properly aligned address
>    - ``error`` the operation will fail with an error

Reviewer: ```suggestion
   - ``warn``: a warning is emitted
   - ``ignore``: nothing, alignment checking is disabled
   - ``reallocate``: the buffer is reallocated to a properly aligned address
   - ``error``: the operation fails with an error
```
